userspace/libsinsp/sinsp.cpp

```diff
@@ -538,17 +561,18 @@ void sinsp::open_common(scap_open_args* oargs)
 	init();
 }
 
-scap_open_args sinsp::factory_open_args(const char* engine_name, scap_mode_t scap_mode)
+scap_open_args sinsp::factory_open_args(const std::string& engine_name, scap_mode_t scap_mode)
 {
 	scap_open_args oargs{};
-	oargs.engine_name = engine_name;
+	oargs.engine_name = engine_name.data();
 	oargs.mode = scap_mode;
 	return oargs;
 }
 
 void sinsp::open_kmod(unsigned long driver_buffer_bytes_dim, const std::unordered_set<uint32_t> &ppm_sc_of_interest, const std::unordered_set<uint32_t> &tp_of_interest)
 {
-	scap_open_args oargs = factory_open_args(KMOD_ENGINE, SCAP_MODE_LIVE);
+	m_engine_name = KMOD_ENGINE;
```

Comment:
Why did you split the string declaration from the `factory_open_args` call?

Reply:
Uh i see, because we are passing a const reference now.

Reply:
and because now sinsp saves the engine name in its state, in this way we can retrieve the running engine from our clients like Falco